lib/Runtime/Library/JavascriptObject.cpp

```diff
@@ -858,10 +859,12 @@ namespace Js
 
         for (uint i = 0; i < ownPropsArray->GetLength(); i++)
         {
-            ownPropsArray->DirectGetItemAt(i, &propKey);
+            BOOL getPropResult = ownPropsArray->DirectGetItemAt(i, &propKey);
+            Assert(getPropResult);
```

Comment:
Good question.  I was assuming that `JavascriptOperators::GetOwnPropertyKeys` always returns a simple JavascriptArray, created by our runtime, so we can make assumptions about it.  However I have not verified this.  @jordonwii is that what you found?

Also I do not know about the assert below for `propertyKeyResult`.  Does `JavascriptConversion::ToPropertyKey` always succeed with a result from `JavascriptOperators::GetOwnPropertyKeys`?